pkg/mco/machineconfig.go

```diff
@@ -327,6 +327,27 @@ func (builder *MCBuilder) WithKernelType(kernelType string) *MCBuilder {
 	return builder
 }
 
+// WithRawConfig sets the specified Config to the MachineConfig.
+func (builder *MCBuilder) WithRawConfig(config []byte) *MCBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	if len(config) == 0 {
+		glog.V(100).Infof("The Config.Raw cannot be empty")
+
+		builder.errorMsg = "'Config.Raw' cannot be empty"
+
+		return builder
+	}
+
+	glog.V(100).Infof("Setting Config.Raw: %v", config)
```

Comment:
Will it be human-readable?

Reply:
It is a string converted to []byte. So it is reversible.

Reply:
I see we have Raw field. Can you share link to documentation of this field

Reply:
In my use case, i want to have ignition config to be passed to machineconfig. Below is code on how i use it.

```
import ignition "github.com/coreos/ignition/v2/config/v3_1/types"
....
var (
	err                       error
	RdmaMachineConfigContents = `

      [Unit]
      Description=RDMA exclusive mode
      Before=kubelet.service crio.service node-valid-hostname.service

      [Service]
      # Need oneshot to delay kubelet
      Type=oneshot
      ExecStart=/usr/bin/bash -c "rdma system set netns exclusive"
      StandardOutput=journal+console
      StandardError=journal+console

      [Install]
      WantedBy=network-online.target`
)

truePointer := true

ignitionconfig := ignition.Config{
	Ignition: ignition.Ignition{
		Version: "3.1.0",
	},
	Systemd: ignition.Systemd{
		Units: []ignition.Unit{
			{
				Enabled:  &truePointer,
				Name:     "rdma.service",
				Contents: &RdmaMachineConfigContents,
			},
		},
	},
}

ignitionconfigraw, err := json.Marshal(ignitionconfig)

rdmaMachineConfig := mco.NewMCBuilder(clients.New(kubeconfigpath), "02-rdma-netns-exclusive-mode").WithLabel("machineconfiguration.openshift.io/role", "workercnf").WithRawConfig(ignitionconfigraw)
```

Reply:
is there any other better way to implement this machine config ?

Reply:
https://docs.openshift.com/container-platform/4.16/rest_api/machine_apis/machineconfig-machineconfiguration-openshift-io-v1.html#spec

Reply:
@trewest 
Agreed. Updated to `glog.V(100).Infof("Setting Config.Raw: %s", string(config))`